src/topicButton.js

```
@@ -1,9 +1,22 @@
 /* exported TopicButton */
-
-const {GObject, Gtk} = imports.gi;
+const {GObject, Gtk, GLib, Gio} = imports.gi;
 
 const {Lockscreen} = imports.lockscreen;
 
+const TopicIface = `
+<node>
+  <interface name="com.endlessm.HackToolbox.topic">
+    <method name="reveal">
+      <arg type="b" direction="in" name="revealed"/>
+      <arg type="b" direction="out" name="revealed"/>
```

Is the out argument necessary? From reading the code I don't see a case where it is going to be different than the in argument.

---

Yeah the only time it would be different is if it somehow failed to set the property.

---

I think in that case the method should throw an exception.